packages/twenty-front/src/modules/app/components/AppRouterProviders.tsx

```diff
@@ -28,16 +28,16 @@ export const AppRouterProviders = () => {
   return (
     <ApolloProvider>
       <ClientConfigProviderEffect />
-      <ClientConfigProvider>
-        <ChromeExtensionSidecarEffect />
-        <ChromeExtensionSidecarProvider>
-          <UserProviderEffect />
-          <UserProvider>
-            <AuthProvider>
-              <ApolloMetadataClientProvider>
-                <ObjectMetadataItemsProvider>
-                  <PrefetchDataProvider>
-                    <AppThemeProvider>
+      <ApolloMetadataClientProvider>
+        <ObjectMetadataItemsProvider>
+          <AppThemeProvider>
```

Comment:
@khuddite I think the AI is right here, one example of a bug that was introduced is the always-loading state here:

<img width="692" alt="Screenshot 2024-11-21 at 14 45 36" src="https://github.com/user-attachments/assets/ed6f5beb-617d-4468-abd6-63bb71bbcb5a">

From a system architecture perspective I think it makes more sense to have the clientConfig stay at the higher-level. It's definitely a tougher problem than it looked initially!

I think what would make the most sense is to introduce a `BaseThemeProvider` at a higher level that would be similar to AppThemeProvider but with a simplified logic just based on system preference. This might be helpful if we start having more logged-out pages...

We can then rename `AppThemeProvider` to `UserThemeProvider` and focus its role on updating the context only based on the WorkspaceMember.

Reply:
In addition to that, I will see if we can move `ErrorBoundary` up to the top level, so the initial config load error gets caught by `ErrorBoundary`(not manually like we do now). If so, we won't even need to introduce any additional states like `isErrored` or `errorMessage`. The position of `ErrorBoundary` wrapper in the current setup doesn't quite make sense to me.

Reply:
The issue to move ErrorBoundary  is that the Sentry config comes from ClientConfig... Maybe one day we should push it during the build process in env-config.js instead but today it won't work I think 🤔

Reply:
That's a good callout, thanks! I could've wasted a lot of time just to figure that out 😁